src/cargo/core/global_cache_tracker.rs

```diff
@@ -1489,6 +1561,13 @@ impl DeferredGlobalLastUse {
         self.mark_git_checkout_used_stamp(git_checkout, None);
     }
 
+    /// Indicates the given [`WorkspaceManifest`] has been used right now.
+    ///
+    /// Also implicitly marks the workspace manifest used, too.
+    pub fn mark_workspace_src_used(&mut self, workspace_src: WorkspaceSrc) {
```

Comment:
Huh, I wonder why we exposed these in the API, rather than taking parameters.  Seems like it creates more boilerplate for the callers